cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -113,17 +111,49 @@ void CrossEntropyLossKernel<algorithmFPType, method, cpu>::softmax(const algorit
         }
     }
     daal::internal::Math<algorithmFPType, cpu>::vExp(nRows * nCols, res, res);
-    for (size_t iRow = 0; iRow < nRows; ++iRow)
+    if (softmaxSums != nullptr && yLocal != nullptr)
```

Comment:
seems `softmaxSums` should be always not equal to `nullptr`.

Reply:
in case intercept == true of course.

Reply:
softmax function is used in other algs (like gbt). Using in places like that doesn't require calculation of softmaxSums array